sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -162,11 +164,15 @@ class GpuParquetFileFormat extends ColumnarFileFormat with Logging {
     val outputTimestampType = sparkSession.sessionState.conf.parquetOutputTimestampType
     if (outputTimestampType != ParquetOutputTimestampType.TIMESTAMP_MICROS) {
```

Comment:
This is really convoluted code now.  Could we just use a match for the output type? and why do we need to set a second config to reflect the value of the first config?

Reply:
+1 we should just port the SQLConf key/value pair into the Hadoop Configuration as Spark does.

Reply:
Still not doing what I asked.

```
sparkSession.sessionState.conf.parquetOutputTimestampType match {
    case ParquetOutputTimestampType.TIMESTAMP_MICROS =>
    case ParquetOutputTimestampType.TIMESTAMP_MILLIS =>
    case outputTimestampType =>
         val hasTimestamps = dataSchema.exists { field =>
            TrampolineUtil.dataTypeExistsRecursively(field.dataType, _.isInstanceOf[TimestampType])
         }
         if (hasTimestamps) {
             throw new UnsupportedOperationException(
                  s"Unsupported output timestamp type: $outputTimestampType")
         }
}
```

Because this code is doing almost exactly the same thing as above we might be able to combine them together.